### ISSUE TITLE
Only have team ID as a field on user, not the whole team object

### ISSUE THREAD

**Issue by oyvinddd** [oyvinddd/officesports#52]
The current `team` property on the Player objects makes it unnecessarily hard to change team names or add properties to team objects. Therefore it makes sense to add a `teamId` (string) property to the player and fetch the correct team in a separate step. The old `team` property should be marked as deprecated until all players have the `teamId` property correctly set, and until all systems handle `teamId` correctly (Android/iOS/web API). If a player does not have a team, the ID should be set to an empty string(or `null`???).

TLDR; add `teamId` to the Player type, deprecate `team` until all players have `teamId` set, then remove it.

- [x] iOS app
- [x] Android app
- [x] Web API

**Comment by boyum** [comment]
Added to API.

**Comment by boyum** [comment]
I've added a few sentences to specify what needs to be done. What should be the default value? `""` or `null`? @oyvinddd

**Comment by oyvinddd** [comment]
I think having `teamId` as an optional and having a default value of `null` is the cleanest approach, @boyum. Currently working on a branch to change this in the iOS code base (WIP).

**Comment by konstantpapp** [comment]
This is now updated on Flutter repo @boyum @oyvinddd . A nullable `teamId` field is added to the `Player` and all new players or updates to profile will be registered according to that.

**Comment by oyvinddd** [comment]
@konstantpapp fyi -- we're adding `teamId` as a field to match objects as well so that we are able to filter historical matches based on said id.
